ansible_collections/arista/avd/plugins/plugin_utils/schema/avddataconverter.py

```diff
@@ -1,234 +1,169 @@
+
+            yield from self.convert_data(data[key], childschema, path + [key])
+
+        # Compile "dynamic_keys"
+        dynamic_keys = {}
+        schema_dynamic_keys = schema.get("dynamic_keys", {})
+        for dynamic_key, childschema in schema_dynamic_keys.items():
+            resolved_keys = get_all(data, dynamic_key)
+            for resolved_key in resolved_keys:
+                dynamic_keys.setdefault(resolved_key, childschema)
+
+        for key, childschema in dynamic_keys.items():
+            if key not in data:
+                # Skip key since there is nothing to convert if the key is not set in data
+                continue
+
+            # Perform type conversion of the data for the child key if required based on "convert_types"
+            if "convert_types" in childschema:
+                oldtype = type(data[key]).__name__
+                data[key] = self.convert_types(childschema["convert_types"], data[key], childschema, path + [key])
+                newtype = type(data[key]).__name__
+                if oldtype != newtype:
+                    path_str = ".".join(path + [key])
+                    yield AvdConversion(key=path_str, oldtype=oldtype, newtype=newtype)
+
+            # Dive in to child keys/schemas
+            yield from self.convert_data(data[key], childschema, path + [key])
+
+    def convert_items(self, items: dict, data: list, schema: dict, path: list):
+        """
+        This function performs conversion on each item with the items subschema
+        """
+        if not isinstance(data, list):
+            return
+
+        for index, item in enumerate(data):
+            # Perform type conversion of the items data if required based on "convert_types"
+            if "convert_types" in items:
+                oldtype = type(data[index]).__name__
+                data[index] = self.convert_types(items["convert_types"], data[index], items, path + [f"[{index}]"])
+                newtype = type(data[index]).__name__
+                if oldtype != newtype:
+                    path_str = ".".join(path + [f"[{index}]"])
+                    yield AvdConversion(key=path_str, oldtype=oldtype, newtype=newtype)
+
+            # Dive in to child items/schema
+            yield from self.convert_data(item, items, path + [f"[{index}]"])
+
+    def convert_types(self, convert_types: list, data, schema: dict, path: list):
+        """
+        This function performs type convertion if necessary on a single data instance.
+        It is invoked for child keys during "keys" conversion and for child items during
+        "items" conversion
+        Returns the converted value is returned to the calling converter.
+        Any convertion errors are ignored and the original value is returned
+        """
+        schema_type = schema.get("type")
+
+        simple_converters = {
+            "str": str,
+            "int": int,
+            "bool": bool,
+        }
```

Comment:
should this be a global var?

Reply:
not sure it makes a difference. We only use it here.

Reply:
fixed